src/prefect/cli/deploy.py

```diff
@@ -1170,9 +1187,19 @@ def _load_deploy_configs_and_actions(ci=False) -> Tuple[List[Dict], Dict]:
         Tuple[List[Dict], Dict]: a tuple of deployment configurations and actions
     """
     try:
-        with open("prefect.yaml", "r") as f:
+        with prefect_file.open("r") as f:
             prefect_yaml_contents = yaml.safe_load(f)
-    except FileNotFoundError:
+    except (FileNotFoundError, IsADirectoryError) as exc:
+        app.console.print(
+            f"Unable to read the specified config file. Reason: {exc}. Skipping.",
+            style="yellow",
+        )
+        prefect_yaml_contents = {}
+    if not isinstance(prefect_yaml_contents, dict):
+        app.console.print(
+            "Unable to parse the specified config file. Skipping.",
+            style="yellow",
+        )
```

Comment:
How does a non-YAML file get handled? Does it raise here?

Reply:
Good catch, I actually had a hard time coming up with something that YAML wouldn't be able to parse. I have something now so Ill add the exception handler for the error and ship a test for it.